File: src/main/java/no/nav/sosialhjelp/soknad/business/db/repositories/sendtsoknad/SendtSoknadRepository.java
package no.nav.sosialhjelp.soknad.business.db.repositories.sendtsoknad;

import no.nav.sosialhjelp.soknad.domain.SendtSoknad;

import java.util.Optional;

public interface SendtSoknadRepository {

    Long opprettSendtSoknad(SendtSoknad sendtSoknad, String eier);
    Optional<SendtSoknad> hentSendtSoknad(String behandlingsId, String eier);
    void oppdaterSendtSoknadVedSendingTilFiks(String fiksforsendelseId, String behandlingsId, String eier);
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/db/repositories/sendtsoknad/SendtSoknadRepository.kt
package no.nav.sosialhjelp.soknad.db.repositories.sendtsoknad

import java.util.Optional

interface SendtSoknadRepository {
    fun opprettSendtSoknad(sendtSoknad: SendtSoknad, eier: String?): Long?
    fun hentSendtSoknad(behandlingsId: String, eier: String?): Optional<SendtSoknad>
    fun oppdaterSendtSoknadVedSendingTilFiks(fiksforsendelseId: String?, behandlingsId: String?, eier: String?)
}
